### PR TITLE
ASTextNode2 to ignore certain text alignments while calculating intrinsic size

### PR DESCRIPTION
**Pull request by nguyenhuy**
ASTextNode2 uses ASTextLayout to calculate its layout and bounding rect. When the constrained width that is used for layout calculation is inf/max (e.g when the node is inside a horizontal stack), ASTextLayout doesn't ignore its right/center/natural text alignment but takes it into account. That results in an unreasonable size (and position).

Fix by detecting when the node is calculating intrinsic size and force its layout alignment to be left. Other alignments should still work when the max width is finite/reasonable.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Source/ASTextNode2.mm [hunk 1]
```diff
@@ -234,9 +234,12 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 
   _textContainer.size = constrainedSize;
   [self _ensureTruncationText];
-  
+
+  // If constrained width is max/inf, the text node is calculating its intrinsic size.
+  const BOOL calculatingIntrinsicSize = (_textContainer.size.width >= ASTextContainerMaxSize.width);
```
Comment (wiseoldduck) [inline Source/ASTextNode2.mm]: Does Texture ever use the `isVerticalForm` property? We should probably just delete it, but if not maybe check for it here.

File: Source/ASTextNode2.mm [hunk 2]
```diff
@@ -321,18 +324,31 @@ - (NSArray *)exclusionPaths
   return _textContainer.exclusionPaths;
 }
 
-- (void)prepareAttributedString:(NSMutableAttributedString *)attributedString
+- (void)prepareAttributedString:(NSMutableAttributedString *)attributedString forIntrinsicSize:(BOOL)isForIntrinsicSize
```
Comment (wiseoldduck) [inline Source/ASTextNode2.mm]: This method name implies the second parameter is a `CGSize`, to me something like `isForIntrinsicSize` would imply a `BOOL`.

**Comment by nguyenhuy** [comment]
Awesome! thank you for reviewing, @Adlai-Holler and @wiseoldduck.

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/37f500c4b791322817aebb1371edf0d6/raw/100ab4bb62037159dd3ecc9749739372163b3b1b/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/f902acd27633bc0ef207150a9a4149af/raw/86a5be3ba5c80a48ba7ecb9fbafd39f6a71335fc/log.txt)

**Pull request merged**